central/image/service/http_handler.go

```diff
@@ -0,0 +1,153 @@
+package service
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"strconv"
+	"time"
+
+	"github.com/pkg/errors"
+	clusterUtil "github.com/stackrox/rox/central/cluster/util"
+	"github.com/stackrox/rox/central/role/sachelper"
+	"github.com/stackrox/rox/generated/storage"
+	"github.com/stackrox/rox/pkg/features"
+	"github.com/stackrox/rox/pkg/httputil"
+	"github.com/stackrox/rox/pkg/images/enricher"
+	"github.com/stackrox/rox/pkg/images/integration"
+	"google.golang.org/grpc/codes"
+)
+
+const (
+	NumberBytes = 100 * 1024
+	timeout     = 10 * time.Minute
+)
+
+type sbomRequestBody struct {
+	Cluster   string `json:"cluster"`
+	ImageName string `json:"image_name"`
+	Force     bool   `json:"force"`
+}
+
+type sbomHttpHandler struct {
+	integration      integration.Set
+	enricher         enricher.ImageEnricher
+	clusterSACHelper sachelper.ClusterSacHelper
+}
+
+var _ http.Handler = (*sbomHttpHandler)(nil)
+
+// SBOMHandler returns a handler for get sbom http request
+func SBOMHandler(integration integration.Set, enricher enricher.ImageEnricher, clusterSACHelper sachelper.ClusterSacHelper) http.Handler {
+	return sbomHttpHandler{
+		integration:      integration,
+		enricher:         enricher,
+		clusterSACHelper: clusterSACHelper,
+	}
+
+}
+
+func (h sbomHttpHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+
+	if r.Method != http.MethodPost {
+		w.WriteHeader(http.StatusMethodNotAllowed)
+		return
+	}
+	// verify scanner v4 is enabled
+	if !features.ScannerV4.Enabled() {
+		httputil.WriteGRPCStyleError(w, codes.NotFound, errors.New("Scanner v4 is not enabled. Enabled scanner v4 to get SBOMs"))
+		return
+	}
+	if !features.SBOMGeneration.Enabled() {
+		httputil.WriteGRPCStyleError(w, codes.NotFound, errors.New("SBOM feature is not enabled"))
+		return
+	}
+	var params sbomRequestBody
+	reqSizeLimit := NumberBytes
+	reqSizeBytes := os.Getenv("ROX_SBOM_API_REQ_SIZE")
+	if reqSizeBytes != "" {
+		reqSizeLimitInt, err := strconv.Atoi(reqSizeBytes)
+		if err == nil {
+			reqSizeLimit = reqSizeLimitInt
+		}
+	}
```

Comment:
Would recommend using `pkg/env` for handling env vars - [which includes](https://github.com/stackrox/stackrox/blob/a26fd772cb732968e2672b3209b9753509639b5d/pkg/env/integersetting.go#L31-L41) `int` conversion logic, validation, and defaulting.  

For example in `pkg/env/image_scan.go` (feel free to create a new file or use this one), add something along the lines of:

```go
SBOMGenerationMaxReqSize = RegisterIntegerSetting("ROX_SBOM_GEN_MAX_REQ_SIZE", 100*1024)
```
And then in the `central/image/service/http_handler.go`:

```go
var (
sbomGenMaxReqSize = env.SBOMGenerationMaxReqSize.IntegerSetting()
)
...
lr := io.LimitReader(r.Body, int64(sbomGenMaxReqSize))
```
or instead of a package level var, may be easier to test by adding it to the handler struct instead:
```go
func SBOMHandler(integration integration.Set, enricher enricher.ImageEnricher, clusterSACHelper sachelper.ClusterSacHelper) http.Handler {
	return sbomHttpHandler{
	   ...
           sbomGenMaxReqSize = env.SBOMGenerationMaxReqSize.IntegerSetting()
	}
}
```

Reply:
done